lib/register.php

```diff
@@ -517,3 +517,133 @@ function gutenberg_revisions_restore( $revisions_data ) {
 	return $revisions_data;
 }
 add_filter( 'wp_prepare_revision_for_js', 'gutenberg_revisions_restore' );
+
+/**
+ * Generates and caches HTML for a URL that contains OpenGraph metadata.
+ *
+ * @param string $url The URL to generate HTML for.
+ * @return mixed OpenGraph HTML, or false if there was no OpenGraph data.
+ */
+function gutenberg_oembed_opengraph( $url ) {
+	$cache_key = 'gutenberg_opengraph_' . md5( $url );
+	$html      = get_transient( $cache_key );
+
+	// Empty string means we tried to get OpenGraph information for this url, but it had none.
+	if ( '' === $html ) {
+		return false;
+	}
+
+	if ( ! empty( $html ) ) {
+		return $html;
+	}
+
+	$get = wp_safe_remote_get(
+		$url,
+		array(
+			'limit_response_size' => 250000,
+			'user-agent'          => 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:20.0) Gecko/20100101 Firefox/20.0',
+		)
+	);
+
+	if ( is_wp_error( $get ) ) {
+		return false;
+	}
+
+	$body = wp_remote_retrieve_body( $get );
+
+	preg_match( '/<title>([^<]+)<\/title>/i', $body, $title_match );
+	$title = empty( $title_match ) ? $url : $title_match[1];
+	preg_match( '/<meta[^>]+property="og:image"[^>]*>/i', $body, $image_match );
+	$image = empty( $image_match ) ? '' : $image_match[0];
+	preg_match( '/content="([^"]+)"/i', $image, $image_match );
+	$image = empty( $image_match ) ? '' : $image_match[1];
+	preg_match( '/<meta[^>]+property="og:description"[^>]*>/i', $body, $desc_match );
+	$desc = empty( $desc_match ) ? '' : $desc_match[0];
+	preg_match( '/content="([^"]+)"/i', $desc, $desc_match );
```

Comment:
I know I kind of quickly put it together like this, but could we maybe make a separate function for this? Like `getOGPropertyContent( $body, $property )`?

Reply:
Good call :) I'd like to have all of those parsed with a single regex if possible, and just return an array of OG data.

Reply:
Sounds good too!